### PR TITLE
Add package manager to package.json

### PR DESCRIPTION
**Pull request by nezouse**
Should be easier to make sure you use correct version of pnpm now.

### CODE REVIEW

**Comment by norswap** [comment]
What's the procedure to update this? Say I update pnpm and do `make install` (`pnpm install -r`), will that automatically update both the lockfile and the version in `package.json`?

File: README.md [hunk 1]
```diff
@@ -15,8 +15,11 @@ Tooling required:
   - Last `foundryup`: 16 May 2023
 - Node.js & [PNPM](https://pnpm.io/) (`npm install -g pnpm`)
   - Tested with Node v20.1.0
+  - If you have any issues while installing dependencies with pnpm you can try to use `corepack` to make sure you use correct version of pnpm.
+    - `corepack enable`
+    - `corepack pnpm install`
```
[the remark this answers is not in the document]
Comment (nezouse) [reply README.md:20]: It's a new feature in nodejs which picks up correct version of package manager from package manager.
So all it does here is it makes sure you will install deps with pnpm@8.6.1 which is specified in package.json

**Comment by nezouse** [comment]
Lockfile will be updated automatically when you install new deps, but version in package.json has to be managed manually

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nezouse) [reply README.md:20]: https://nodejs.org/api/corepack.html

**Comment by norswap** [comment]
Is there a way to modify `make install` such that in addition to updating the lockfile, it also updates the `packageManager` property?

If that's too hard, we could add `corepack enable` to `make setup` then change all `pnpm` commands in the makefile to use `corepack pnpm` instead?

**Comment by norswap** [comment]
The scenario I'm thinking of is that right now if I (1) update pnpm, (2) run `make install` then push, then the version is not correct anymore (mismatch between `pnpm-lock.json` and `packageManager` & even someone using corepack will get an error.

Unless pnpm itself checks `packageManager` & prevents this from happening?

**Comment by norswap** [comment]
Ok, so I think this is good enough for now. Thinking of it, there is no trivial solution anyway, since the lock file version doesn't update with every version of pnpm. We'll just have to be vigilant to upgrade the statement in the package file every time the lock-file version changes.

I've added warnings to the Makefile and that'll have to do!

**Comment by norswap** [comment]
I completely missed that this also was setting up Vite. This was not fully integrated in the build flow & needs further review, so I've reverted on master.

**Pull request merged**